### PR TITLE
feat: Initial extraction to common lib

### PR DESCRIPTION
**Pull request by piotrgrundas**
Extracts the Saleor-app runtime out of `apps/stripe` into shared packages, so the next
app starts from verified plumbing instead of a copy. First step of NIM-64; `apps/stripe`
is the reference implementation and its behaviour is unchanged.

Every Saleor app we run hand-rolls the same server plumbing — manifest, register endpoint,
webhook signature verification, JWKS, env parsing, the Dashboard iframe shell, the vite
build. The sitemap generator, the Algolia sync app and the mailer each carry an older copy
of what `apps/stripe` holds the newest generation of; ~55 files are byte-identical between
two of them. The cost is not the duplication, it is the drift: a fix to webhook
verification lands in one app and is invisible to the rest.

## Two packages, not one

- **`@nimara/lib`** — runtime. Bundled into every app that uses it.
- **`@nimara/tooling`** — build-time only. Zero runtime dependencies, `vite` as a peer.

The split is not cosmetic. `@nimara/tooling` is loaded by Node while `vite.config.ts` is
read, before any `@/*` alias exists — hence the literal `.ts` extensions in its relative
imports. Folding it into the runtime package would put vite and its plugins in the graph
of every consumer.

### What moved into `@nimara/lib`

`src/error` + `src/api/schema` (HTTP error hierarchy, one response envelope) · `src/config`
(`baseConfigSchema(pkg)`, app identity derived from the consumer's `package.json`) ·
`src/saleor` (URL/domain helpers, allowlist, webhook registry, tenant) · `src/hono/**`
(middleware, error handler, manifest + register + webhook routes) · `src/client/**`
(Dashboard shell, app-bridge handshake) · `src/test` (harness).

Everything that names Hono lives under `src/hono/**`. What is left outside it is what
another runtime could reuse unchanged — which is what makes a queue-driven app possible
later.

### What moved into `@nimara/tooling`

Per-app entry discovery under `src/apps/*`, the shared vite config factory, and one build
adapter per deployment target (`node`, `vercel`) behind a single `BuildTargetAdapter` seam.

## Security fixes folded in

**Cross-tenant writes.** A caller holding a valid token or webhook signature for *one*
Saleor could act on *another* installation, because handlers took the domain from the
request body. The tenant is now established only by a middleware that verified a signature
against that installation's JWKS, published on the context, and handed to webhook handlers
as an argument — a handler cannot be written that forgets to scope by it. Regression tests
cover both the token and the webhook path.

**Unauthenticated Dashboard API.** `/api/app/config/*` now sits behind
`saleorTokenMiddleware({ requiredPermissions: ["MANAGE_APPS"] })` plus
`saleorTenantMiddleware()`. Without a required permission any token Saleor ever signed
would have passed, including an ordinary storefront customer's.

**One allowlist.** `createSaleorRoutes` took `allowedDomains` twice — once for install,
once for webhooks. There is no case where installation admits one set of domains and
webhooks another, so it is one parameter mounted in front of both.

## Behaviour changes worth a reviewer's attention

- **Env parsing is stricter.** A variable left blank no longer reads as unset, and
  `SENTRY_DSN` is validated as a URL. Blank `SENTRY_DSN=` / `VITE_SALEOR_APP_TOKEN=` lines
  must be commented out or the app will not boot. Previously a mistyped DSN silently
  disabled reporting — Sentry does not throw on a bad DSN, it disables itself.
- **`VITE_SALEOR_APP_TOKEN` is refused in production.** It is a development-only token.
- **`.gitignore`** carried a bare `build` pattern that hid `packages/tooling/src/build/`
  entirely — source, not output. Negated for that path only; `apps/docs/build` stays
  ignored.
- **`packages/config/src/.fallowrc.json`** gains `lib` and `tooling` zones. Both report
  zero boundary violations, and the seven pre-existing `stripe -> foundation` violations
  are gone.

## Testing

`pnpm test` — 8/8 turbo tasks. `@nimara/lib` 92 tests, `stripe` 116, all green.
`lint:types` clean for `lib`, `tooling` and `stripe`; Prettier clean; `fallow` reports
`unresolved_imports: 0`, `unlisted_dependencies: 0`.

## Risk and rollback

`apps/stripe` is the only consumer, so the blast radius is one app. Both deployment
targets build (`BUILD_TARGET=node` and `vercel`). Rollback is reverting the merge — no
data migration, no config migration beyond the `.env` lines noted above.

# Pull Request Checklist

- [ ] Target `main` from a short-lived change branch and use a Conventional Commit PR title.
- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Verify all four Vercel project statuses: `nimara-docs`, `nimara-ecommerce`,
      `nimara-ecommerce-stripe`, and `nimara-marketplace`.
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Keep incomplete behavior disabled with a short-lived feature flag or branch-by-abstraction seam.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

**Review by lukas-szewczyk — APPROVED** [review]

File: apps/stripe/src/apps/handler/api/rest/saleor/index.ts [hunk 1]
```diff
@@ -18,149 +16,111 @@ import {
...
+  transactionProcessSessionHandler,
+  transactionRefundRequestedHandler,
+} from "./webhooks/transactions";
 
-        return responseFromErrors(result.errors);
-      }
+const CONFIG = container.get("config");
 
-      logger.info(`Installation successful for ${saleorDomain}.`);
+const webhooks: SaleorWebhook<HandlerContext<any>>[] = [
```
Comment (gajda-w) [inline apps/stripe/src/apps/handler/api/rest/saleor/index.ts:37]: maybe worth type this instead of `any` ?

**Review by gajda-w — APPROVED** [review]
